docs/code/IDataViewDesignPrinciples.md

```diff
@@ -0,0 +1,479 @@
+  cursors on the view produce the same row values in the same order (when
+  using the same shuffling). See [here](#immutability-and-repeatability).
+
+* **Memory efficiency**: The IDataView design includes cooperative buffer
+  sharing patterns that eliminate the need to allocate objects or buffers for
+  each row when cursoring through a view. See [here](#memory-efficiency).
+
+* **Batch-parallel computation**: The IDataView system includes the ability to
+  get a set of cursors that can be executed in parallel, with each individual
+  cursor serving up a subset of the rows. Splitting into multiple cursors can
+  be done either at the loader level or at an arbitrary point in a pipeline.
+  The component that performs splitting also provides the consolidation logic.
+  This enables computation heavy pipelines to leverage multiple cores without
+  complicating each individual transform implementation. See
+  [here](#batch-parallel-cursoring).
+
+* **Large data support**: Constructing views on data files and cursoring
+  through the rows of a view does not require the entire data to fit in
+  memory. Conversely, when the entire data fits, there is nothing preventing
+  it from being loaded entirely in memory. See [here](#data-size).
+
+### Design Non-requirements
+
+The IDataView system design does *not* include the following:
+
+* **Multi-view schema information**: There is no direct support for specifying
+  cross-view schema information, for example, that certain columns are primary
+  keys, and that there are foreign key relationships among tables. However,
+  the column metadata support, together with conventions, may be used to
+  represent such information.
+
+* **Standard ML schema**: The IDataView system does not define, nor prescribe,
+  standard ML schema representation. For example, it does not dictate
+  representation of nor distinction between different semantic interpretations
+  of columns, such as label, feature, score, weight, etc. However, the column
+  metadata support, together with conventions, may be used to represent such
+  interpretations.
+
+* **Row count**: A view is not required to provide its row count. The
+  `IDataView` interface has a `GetRowCount` method with type `Nullable<long>`.
+  When this returns `null`, the row count is not available directly from the
+  view.
+
+* **Efficient indexed row access**: There is no standard way in the IDataView
+  system to request the values for a specific row number. While the
+  `IRowCursor` interface has a `MoveMany(long count)` method, it only supports
+  moving forward `(count > 0)`, and is not necessarily more efficient than
+  calling `MoveNext()` repeatedly. See [here](#row-cursor).
+
+* **Data file formats**: The IDataView system does not dictate storage or
+  transport formats. It *does* include interfaces for loader and saver
+  components. The AzureML Algorithms team has implemented loaders and savers
```

Comment:
There's a couple of references to "AzureML Algorithms team", should this remain?